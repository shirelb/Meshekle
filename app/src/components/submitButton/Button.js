import React, {Component} from 'react';
import {StyleSheet, Text, TouchableOpacity} from "react-native";
import PropTypes from 'prop-types'
import colors from "../../shared/colors";


export default class Button extends Component {
    render() {
        const {label, onPress} = this.props;
        return (
            <TouchableOpacity style={Object.assign({},
                styles.container,
                {backgroundColor: this.props.color ? this.props.color : colors.DODGER_BLUE})}
                              onPress={onPress}>
                <Text style={styles.text}>{label}</Text>
            </TouchableOpacity>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        width: "100%",
        alignItems: "center",
        justifyContent: "center",
        marginBottom: 12,
        paddingVertical: 12,
        borderRadius: 4,
        borderWidth: StyleSheet.hairlineWidth,
        borderColor: "rgba(255,255,255,0.7)"
    },
    // containerBGcolor: {
    //     backgroundColor: colors.DODGER_BLUE,
    // },
    text: {
        color: colors.WHITE,
        textAlign: "center",
        height: 20
    }
});

Button.propTypes = {
    label: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
};