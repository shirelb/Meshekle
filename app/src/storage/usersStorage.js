import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getUserEvents = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/events/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user events error ', error)
        });
};

var getUserById = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var getUsers = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/users`,
        {headers: userHeaders}
    )
        .then(response => {
            return response.data.filter(user => user.userId !== 1 && user.userId !== "1");
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var userValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/users/validToken`,
        {
            "token": token,
        },
    );
}

var userLogin = function (userId, password) {
    return axios.post(`${SERVER_URL}/api/users/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
}


export default {getUserEvents, getUserById, userValidToken, userLogin, getUsers};
