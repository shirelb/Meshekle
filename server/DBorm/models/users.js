module.exports = (sequelize, type) => {
    return sequelize.define('Users', {
        userId:{
            type: type.STRING,
            primaryKey: true,
            required: true
        },
        fullname: {
            type: type.STRING,
            required: true
        },
        password: {
            type: type.STRING,
            required: true
        },
        email: {
            type: type.STRING,
            required: true
        },
        mailbox: {
            type: type.INTEGER,
            required: true
        },
        cellphone: {
            type: type.STRING,
            required: true
        },
        phone: {
            type: type.STRING
        },
        bornDate: {
            type: type.DATE,
            required: true
        },
        active: {
            type: type.BOOLEAN,
            required: true,
            defaultValue: true
        },
        image: {
            type: type.BLOB,
        }
    });
};