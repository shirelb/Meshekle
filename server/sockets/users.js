
module.exports = function (socket,pushToAllWebClient,pushToAllAppClient) {
    socket.on('userShallowDeleted', (data) => {
        console.log('socket userShallowDeleted data ', data);
        pushToAllWebClient('getUsers');
        pushToAllAppClient('getUsers');
    });

    socket.on('userCreated', (data) => {
        console.log('socket userCreated data ', data);
        pushToAllWebClient('getUsers');
        pushToAllAppClient('getUsers');
    });

    socket.on('userUpdated', (data) => {
        console.log('socket userUpdated data ', data);
        pushToAllWebClient('getUsers');
        pushToAllAppClient('getUsers');
    });


};
