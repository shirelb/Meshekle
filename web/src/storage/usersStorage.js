import axios from "axios";
import {SERVER_URL, WEB_SOCKET} from "../shared/constants";
import store from "store";

const getServiceProviderHeaders = () => {
    return {
        'Authorization': 'Bearer ' + store.get('serviceProviderToken')
    };
};

var getUserByUserID = (userId, headers = null) => {
    return axios.get(`${SERVER_URL}/api/users/userId/${userId}`,
        {headers: headers ? headers : getServiceProviderHeaders()}
    )
        .then((response) => {
            return response.data[0];
        })
        .catch((error) => {
            console.log('getUserByUserID ', userId, ' error ', error);
            return error;
        });
};

var getUsers = (headers = null) => {
    return axios.get(`${SERVER_URL}/api/users`,
        {headers: headers ? headers : getServiceProviderHeaders()}
    )
        .then((response) => {
            return response.data.filter(user => user.userId !== 1 && user.userId !== "1");
        })
        .catch((error) => {
            console.log('getUsers error ', error);
            return error;
        });
};

var deleteUserByUserID = (userId, headers = null) => {
    return axios.delete(`${SERVER_URL}/api/serviceProviders/users/userId/${userId}/delete`,
        {headers: headers ? headers : getServiceProviderHeaders()}
    )
        .then((response) => {
            WEB_SOCKET.emit('userShallowDeleted');

            return response;
        })
        .catch((error) => {
            console.log('deleteUserByUserID ', userId, ' error ', error);
            return error;
        });
};

var createUser = (newUser, serviceProviderHeaders) => {
    serviceProviderHeaders['Access-Control-Allow-Origin'] = '*';
    return axios.post(`${SERVER_URL}/api/serviceProviders/users/add`,
        {
            userId: newUser.userId,
            fullname: newUser.fullname,
            email: newUser.email,
            mailbox: newUser.mailbox,
            cellphone: newUser.cellphone,
            phone: newUser.phone,
            bornDate: newUser.bornDate,
            image: newUser.image ? newUser.image : null,
        },
        {headers: serviceProviderHeaders}
    )
        .then((response) => {
            WEB_SOCKET.emit('userCreated');

            return response;
        })
        .catch((error) => {
            console.log('createUser error ', error);
            return error;
        });
};

var updateUserById = function (updatedUser, serviceProviderHeaders) {
    return axios.put(`${SERVER_URL}/api/users/update/userId/${updatedUser.userId}`,
        {
            fullname: updatedUser.fullname ? updatedUser.fullname : null,
            password: updatedUser.password ? updatedUser.password : null,
            email: updatedUser.email ? updatedUser.email : null,
            mailbox: updatedUser.mailbox ? updatedUser.mailbox : null,
            cellphone: updatedUser.cellphone ? updatedUser.cellphone : null,
            phone: updatedUser.phone ? updatedUser.phone : null,
            bornDate: updatedUser.bornDate ? updatedUser.bornDate : null,
            active: typeof updatedUser.active === 'boolean' ? updatedUser.active : null,
            image: updatedUser.image ? updatedUser.image : null,
        },
        {headers: serviceProviderHeaders}
    )
        .then(response => {
            WEB_SOCKET.emit('userUpdated');

            return response;
        })
        .catch(error => {
            console.log('updateUserById error ', error);
            return error;
        });
};


export default {getUserByUserID, getUsers, deleteUserByUserID, createUser, updateUserById};
