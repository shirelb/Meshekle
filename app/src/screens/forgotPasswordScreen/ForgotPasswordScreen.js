import React, {Component} from 'react';
import {Alert, ScrollView, StyleSheet, TextInput, View} from "react-native";
import {FormLabel, FormValidationMessage, Text} from "react-native-elements";
import DateTimePicker from 'react-native-modal-datetime-picker';
import Button from "../../components/submitButton/Button";
import moment from 'moment';
import usersStorage from "../../storage/usersStorage";
import colors from "../../shared/colors";


export default class ForgotPasswordScreen extends Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {},

            errorMsg: '',
            errorVisible: false
        };
    }

    // componentWillReceiveProps(nextProps, nextContext) {
    //     this.setState({
    //         isDateTimePickerVisible: false,
    //         errorMsg: '',
    //         errorVisible: true
    //     });
    // }

    validateForm = () => {
        let user = this.state.user;
        if (user.userId === '' || !(/^\d*$/.test(user.userId))) {
            this.setState({
                formError: true,
                errorMsg: "ת.ז. חסר וצריך להכיל רק ספרות",
                errorVisible: true
            });
            return false;
        }

        if (user.email === '' || !(/^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/.test(user.email))) {
            this.setState({
                formError: true,
                errorMsg: "אימייל חסר או לא וואלידי",
                errorVisible: true
            });
            return false;
        }

        if (!user.mailbox || !(/^\d*$/.test(user.mailbox))) {
            this.setState({
                formError: true,
                errorMsg: "תיבת דואר חסרה וצריכה להכיל רק ספרות",
                errorVisible: true
            });
            return false;
        }

        if (!user.cellphone && !user.phone) {
            this.setState({
                formError: true,
                errorMsg: "עלייך למלא פלאפון או טלפון",
                errorVisible: true
            });
            return false;
        }

        if (user.cellphone && !(/^\d*$/.test(user.cellphone))) {
            this.setState({
                formError: true,
                errorMsg: "הפלאפון לא וואלידי",
                errorVisible: true
            });
            return false;
        }

        if (user.phone && !(/^\d*$/.test(user.phone))) {
            this.setState({
                formError: true,
                errorMsg: "הטלפון לא וואלידי",
                errorVisible: true
            });
            return false;
        }

        if (!user.bornDate || user.bornDate === null) {
            this.setState({
                formError: true,
                errorMsg: "תאריך הלידה חסר",
                errorVisible: true
            });
            return false;
        }

        return true;
    };

    handleForgetPassword = (e) => {
        if (!this.validateForm()) {
            return;
        }

        usersStorage.forgetPassword(this.state.user)
            .then(response => {
                console.log('forgetPassword response ', response);
                if (!response) {
                    this.setState({showError: true});
                } else {
                    Alert.alert(
                        'התראה',
                        'במידה והפרטים נכונים תישלח אלייך סיסמא חדשה למייל. בבקשה לשנות אותה ברגע כניסה לאפליקציה',
                    );
                    this.props.navigation.navigate('LoginScreen');
                }
            })
            .catch(response => {
                this.setState({showError: true});
            })
    };


    render() {
        let user = this.state.user;

        return (
            <View style={{marginTop: 20}}>
                <ScrollView>

                    <Text h4 style={styles.textTitle}>שכחתי סיסמא</Text>


                    <View style={{flex: 1, flexDirection: 'row', justifyContent: 'space-between',}}>
                        <FormLabel labelStyle={styles.formText}>*ת.ז.</FormLabel>
                        <TextInput
                            value={user.userId}
                            // placeholder={user.email}
                            onChangeText={(text) => this.setState({
                                user: {
                                    ...this.state.user,
                                    userId: text
                                }
                            })}
                            onFocus={() => this.setState({errorVisible: false, showError: false, errorMsg: ''})}
                            underlineColorAndroid="#4aba91"
                            style={styles.textInput}
                            // autoComplete={"email"}
                            keyboardType={"number-pad"}
                            // textContentType={'emailAddress'}
                        />
                    </View>

                    <View style={{flex: 1, flexDirection: 'row', justifyContent: 'space-between',}}>
                        <FormLabel labelStyle={styles.formText}>*אימייל</FormLabel>
                        <TextInput
                            value={user.email}
                            // placeholder={user.email}
                            onChangeText={(text) => this.setState({
                                user: {
                                    ...this.state.user,
                                    email: text
                                }
                            })}
                            onFocus={() => this.setState({errorVisible: false, showError: false, errorMsg: ''})}
                            underlineColorAndroid="#4aba91"
                            style={styles.textInput}
                            autoComplete={"email"}
                            keyboardType={"email-address"}
                            textContentType={'emailAddress'}
                        />
                    </View>

                    <View style={{flex: 1, flexDirection: 'row', justifyContent: 'space-between',}}>
                        <FormLabel labelStyle={styles.formText}> תיבת דואר</FormLabel>
                        <TextInput
                            value={user.mailbox ? user.mailbox.toString() : ""}
                            // placeholder={user.mailbox ? user.mailbox.toString() : ""}
                            onChangeText={(text) => this.setState({
                                user: {
                                    ...this.state.user,
                                    mailbox: text
                                }
                            })}
                            onFocus={() => this.setState({errorVisible: false, showError: false, errorMsg: ''})}
                            underlineColorAndroid="#4aba91"
                            style={styles.textInput}
                            keyboardType={"number-pad"}
                        />
                    </View>

                    <View style={{flex: 1, flexDirection: 'row', justifyContent: 'space-between',}}>
                        <FormLabel labelStyle={styles.formText}>פלאפון</FormLabel>
                        <TextInput
                            value={user.cellphone}
                            // placeholder={user.cellphone}
                            onChangeText={(text) => this.setState({
                                user: {
                                    ...this.state.user,
                                    cellphone: text
                                }
                            })}
                            onFocus={() => this.setState({errorVisible: false, showError: false, errorMsg: ''})}
                            underlineColorAndroid="#4aba91"
                            style={styles.textInput}
                            autoComplete={"tel"}
                            keyboardType={"phone-pad"}
                            textContentType={'telephoneNumber'}
                        />
                    </View>

                    <View style={{flex: 1, flexDirection: 'row', justifyContent: 'space-between',}}>
                        <FormLabel labelStyle={styles.formText}> טלפון</FormLabel>
                        <TextInput
                            value={user.phone}
                            // placeholder={user.phone}
                            onChangeText={(text) => this.setState({
                                user: {
                                    ...this.state.user,
                                    phone: text
                                }
                            })}
                            onFocus={() => this.setState({errorVisible: false, showError: false, errorMsg: ''})}
                            underlineColorAndroid="#4aba91"
                            style={styles.textInput}
                            autoComplete={"tel"}
                            keyboardType={"phone-pad"}
                            textContentType={'telephoneNumber'}
                        />
                    </View>

                    <View style={{flex: 1, flexDirection: 'row', justifyContent: 'flex-start',}}>
                        <FormLabel labelStyle={styles.formText}> תאריך לידה</FormLabel>
                        <TextInput
                            value={user.bornDate ? moment(user.bornDate).format("DD/MM/YYYY") : ""}
                            // placeholder={moment(user.bornDate).format("DD/MM/YYYY")}
                            onFocus={() => this.setState({
                                errorVisible: false,
                                isDateTimePickerVisible: true,
                                showError: false
                            })}
                            underlineColorAndroid="#4aba91"
                            style={styles.textInput}
                        />
                        <DateTimePicker
                            isVisible={this.state.isDateTimePickerVisible}
                            onConfirm={(date) => this.setState({
                                user: {
                                    ...this.state.user,
                                    bornDate: moment(date).format()
                                },
                                isDateTimePickerVisible: false
                            })}
                            onCancel={() => this.setState({isDateTimePickerVisible: false})}
                            is24Hour={true}
                            mode={'date'}
                            datePickerModeAndroid={'spinner'}
                            title='תאריך לידה'
                        />
                    </View>

                    {
                        this.state.errorVisible === true ?
                            <FormValidationMessage>{this.state.errorMsg}</FormValidationMessage>
                            : null
                    }

                    <View style={{marginTop: 20}}>
                        <Button
                            label='שחזר סיסמא'
                            onPress={this.handleForgetPassword}
                        />

                        <Button
                            label='חזור'
                            onPress={() => {
                                this.props.navigation.navigate('LoginScreen');
                            }}
                        />
                    </View>

                    {this.state.showError ?
                        <View>
                            <Text style={styles.errorText}>
                                ישנם פרטים שאינם נכונים לפי המידע השמור במערכת או שהמשתמש אינו קיים
                            </Text>
                        </View>
                        : null
                    }

                </ScrollView>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    errorText: {
        textAlign: 'center',
        color: colors.TORCH_RED,
    },

    textTitle: {
        textAlign: 'center',
        color: '#050505',
        fontWeight: 'bold',
        marginBottom: 20,
    },

    formText: {
        fontSize: 16
    },

    formInputContainer: {
        borderBottomWidth: 1,
        width: 180,
        marginRight: 40,
    },

    textInput: {
        height: 50,
        width: 200,
        // borderRadius: 10 ,
        // borderWidth: 2,
        // borderColor: '#009688',
        backgroundColor: "#FFF",
        // marginBottom: 10,
        // color: '#bbbbbb',
        fontSize: 16,
        // textAlign: 'center',
        marginLeft: 20,
        marginRight: 20,
    }
});
