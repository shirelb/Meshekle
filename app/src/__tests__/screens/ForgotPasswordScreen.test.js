import React from 'react';

import moment from 'moment';

import {ScrollView, TextInput} from "react-native";
import Button from "../../components/submitButton/Button";
import {FormLabel, FormValidationMessage, Text} from "react-native-elements";

import {shallow} from "enzyme/build";

import ForgotPasswordScreen from "../../screens/forgotPasswordScreen/ForgotPasswordScreen";
import usersStorage from "../../storage/usersStorage";
import phoneStorage from "react-native-simple-store";
import users from "../jsons/users";

jest.mock("../../storage/serviceProvidersStorage");
jest.mock("../../storage/usersStorage");
jest.mock("react-native-simple-store");


describe("ForgotPasswordScreen should", () => {
    let wrapper = null;
    let componentInstance = null;
    const props = {};
    const userTest = users[2];
    const userIdTest = "972350803";
    const userFullnameTest = "Dion Revance";

    const mockStore = {
        userData: {
            token: "some token",
            userId: userIdTest,
            userFullname: userFullnameTest,
        }
    };

    const navigation = {navigate: jest.fn()};
    usersStorage.getUsers = jest.fn().mockResolvedValue(users);
    phoneStorage.get = jest.fn().mockImplementation((key) => Promise.resolve(mockStore[key]));
    phoneStorage.update = jest.fn().mockImplementation((key, value) => Promise.resolve(mockStore[key] = value));

    beforeEach(() => {
        jest.clearAllMocks();
    });

    xit('match snapshot', async () => {
        wrapper = shallow(<ForgotPasswordScreen/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("render what the user see", async () => {
        wrapper = shallow(<ForgotPasswordScreen/>);
        componentInstance = wrapper.instance();

        expect(wrapper.find(ScrollView)).toHaveLength(1);
        expect(wrapper.find(Text)).toHaveLength(1);
        expect(wrapper.find(Text).props().children).toEqual("שכחתי סיסמא");
        expect(wrapper.find('FormLabel')).toHaveLength(6);
        expect(wrapper.find('FormLabel').at(0).props().children).toEqual('*ת.ז.');
        expect(wrapper.find('FormLabel').at(1).props().children).toEqual('*אימייל');
        expect(wrapper.find('FormLabel').at(2).props().children).toEqual(' תיבת דואר');
        expect(wrapper.find('FormLabel').at(3).props().children).toEqual('פלאפון');
        expect(wrapper.find('FormLabel').at(4).props().children).toEqual(' טלפון');
        expect(wrapper.find('FormLabel').at(5).props().children).toEqual(' תאריך לידה');
        expect(wrapper.find('TextInput')).toHaveLength(6);
        expect(wrapper.find('TextInput').at(0).props().value).toBeUndefined();
        expect(wrapper.find('TextInput').at(1).props().value).toBeUndefined();
        expect(wrapper.find('TextInput').at(2).props().value).toEqual('');
        expect(wrapper.find('TextInput').at(3).props().value).toBeUndefined();
        expect(wrapper.find('TextInput').at(4).props().value).toBeUndefined();
        expect(wrapper.find('TextInput').at(5).props().value).toEqual('');
        expect(wrapper.find('Button')).toHaveLength(2);
        expect(wrapper.find('Button').at(0).props().label).toEqual('שחזר סיסמא');
        expect(wrapper.find('Button').at(1).props().label).toEqual('חזור');

        expect(componentInstance.state.user).toEqual({});
        expect(componentInstance.state.errorMsg).toEqual('');
        expect(componentInstance.state.errorVisible).toEqual(false);
    });

    it("handle change of userId", async () => {
        wrapper = shallow(<ForgotPasswordScreen/>);
        componentInstance = wrapper.instance();

        await wrapper.find('TextInput').at(0).props().onChangeText(userIdTest);
        expect(componentInstance.state.user.userId).toEqual(userIdTest);
    });

    it("handle change of email", async () => {
        wrapper = shallow(<ForgotPasswordScreen/>);
        componentInstance = wrapper.instance();

        await wrapper.find('TextInput').at(1).props().onChangeText(userTest.email);
        expect(componentInstance.state.user.email).toEqual(userTest.email);
    });

    it("validate form of forgot password", async () => {
        wrapper = shallow(<ForgotPasswordScreen/>);
        componentInstance = wrapper.instance();

        componentInstance.setState({
            user: {
                "userId": "972350d803",
            }
        });
        let validateFormResponse = componentInstance.validateForm();
        expect(validateFormResponse).toEqual(false);
        expect(componentInstance.state.errorMsg).toEqual("ת.ז. חסר וצריך להכיל רק ספרות");
        expect(componentInstance.state.errorVisible).toEqual(true);

        componentInstance.setState({
            user: {
                "userId": userIdTest,
                "email": "972350d803",
            }
        });
        validateFormResponse = componentInstance.validateForm();
        expect(validateFormResponse).toEqual(false);
        expect(componentInstance.state.errorMsg).toEqual("אימייל חסר או לא וואלידי");
        expect(componentInstance.state.errorVisible).toEqual(true);

        componentInstance.setState({
            user: {
                "userId": userIdTest,
                email: userTest.email,
                "mailbox": "972350d803",
            }
        });
        validateFormResponse = componentInstance.validateForm();
        expect(validateFormResponse).toEqual(false);
        expect(componentInstance.state.errorMsg).toEqual("תיבת דואר חסרה וצריכה להכיל רק ספרות");
        expect(componentInstance.state.errorVisible).toEqual(true);

        componentInstance.setState({
            user: {
                "userId": userIdTest,
                email: userTest.email,
                mailbox: userTest.mailbox,
            }
        });
        validateFormResponse = componentInstance.validateForm();
        expect(validateFormResponse).toEqual(false);
        expect(componentInstance.state.errorMsg).toEqual("עלייך למלא פלאפון או טלפון");
        expect(componentInstance.state.errorVisible).toEqual(true);

        componentInstance.setState({
            user: {
                "userId": userIdTest,
                email: userTest.email,
                mailbox: userTest.mailbox,
                "phone": "dddddd",
            }
        });
        validateFormResponse = componentInstance.validateForm();
        expect(validateFormResponse).toEqual(false);
        expect(componentInstance.state.errorMsg).toEqual("הטלפון לא וואלידי");
        expect(componentInstance.state.errorVisible).toEqual(true);

        componentInstance.setState({
            user: {
                "userId": userIdTest,
                email: userTest.email,
                mailbox: userTest.mailbox,
                phone: userTest.phone,
                "bornDate": null,
            }
        });
        validateFormResponse = componentInstance.validateForm();
        expect(validateFormResponse).toEqual(false);
        expect(componentInstance.state.errorMsg).toEqual("תאריך הלידה חסר");
        expect(componentInstance.state.errorVisible).toEqual(true);

        componentInstance.setState({
            errorMsg: "",
            errorVisible: false,
            user: {
                "userId": "972350803",
                "email": "drevance2@google.fr",
                "mailbox": 3,
                "cellphone": "0538274431",
                "phone": "041748647",
                "bornDate": moment("1966-07-30T10:19:42.000Z"),
            }
        });
        validateFormResponse = componentInstance.validateForm();
        expect(validateFormResponse).toEqual(true);
        expect(componentInstance.state.errorMsg).toEqual("");
        expect(componentInstance.state.errorVisible).toEqual(false);

    });

    it("submit forgot password form with wrong data", async () => {
        wrapper = shallow(<ForgotPasswordScreen/>);
        componentInstance = wrapper.instance();
        const handleForgetPasswordSpy = jest.spyOn(componentInstance, 'handleForgetPassword');

        await componentInstance.setState({user: {userId: "7j"}});
        await wrapper.find('Button').at(0).props().onPress();

        expect(handleForgetPasswordSpy).toHaveBeenCalled();
        expect(componentInstance.state.errorMsg).toEqual("ת.ז. חסר וצריך להכיל רק ספרות");
        expect(componentInstance.state.errorVisible).toEqual(true);
        expect(wrapper.find('FormValidationMessage')).toHaveLength(1);
        expect(wrapper.find('FormValidationMessage').props().children).toEqual("שגיאה בעת מילוי טופס:\n" +
            "ת.ז. חסר וצריך להכיל רק ספרות");
    });

    it("submit forgot password form with right data", async () => {
        usersStorage.forgetPassword = jest.fn().mockResolvedValue('response of usersStorage forgot password');

        wrapper = shallow(<ForgotPasswordScreen navigation={navigation}/>);
        componentInstance = wrapper.instance();
        const handleForgetPasswordSpy = jest.spyOn(componentInstance, 'handleForgetPassword');

        await componentInstance.setState({user: userTest});
        // const validateFormSpy = jest.spyOn(componentInstance, 'validateForm');
        // validateFormSpy.mockReturnValue(true);

        await wrapper.find(Button).at(0).props().onPress();

        expect(handleForgetPasswordSpy).toHaveBeenCalled();

        expect(wrapper.find(FormValidationMessage)).toHaveLength(0);
        expect(componentInstance.state.errorMsg).toEqual('');
        expect(componentInstance.state.errorVisible).toEqual(false);
    });
});