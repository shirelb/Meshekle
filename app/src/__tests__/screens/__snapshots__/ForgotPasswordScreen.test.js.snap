// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordScreen should match snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <ScrollViewMock>
    <TextElement
      h4={true}
      style={
        Object {
          "color": "#050505",
          "fontWeight": "bold",
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      שכחתי סיסמא
    </TextElement>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <FormLabel
        labelStyle={
          Object {
            "fontSize": 16,
          }
        }
      >
        *ת.ז.
      </FormLabel>
      <TextInput
        allowFontScaling={true}
        keyboardType="number-pad"
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 16,
            "height": 50,
            "marginLeft": 20,
            "marginRight": 20,
            "width": 200,
          }
        }
        underlineColorAndroid="#4aba91"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <FormLabel
        labelStyle={
          Object {
            "fontSize": 16,
          }
        }
      >
        *אימייל
      </FormLabel>
      <TextInput
        allowFontScaling={true}
        autoComplete="email"
        keyboardType="email-address"
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 16,
            "height": 50,
            "marginLeft": 20,
            "marginRight": 20,
            "width": 200,
          }
        }
        textContentType="emailAddress"
        underlineColorAndroid="#4aba91"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <FormLabel
        labelStyle={
          Object {
            "fontSize": 16,
          }
        }
      >
         תיבת דואר
      </FormLabel>
      <TextInput
        allowFontScaling={true}
        keyboardType="number-pad"
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 16,
            "height": 50,
            "marginLeft": 20,
            "marginRight": 20,
            "width": 200,
          }
        }
        underlineColorAndroid="#4aba91"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <FormLabel
        labelStyle={
          Object {
            "fontSize": 16,
          }
        }
      >
        פלאפון
      </FormLabel>
      <TextInput
        allowFontScaling={true}
        autoComplete="tel"
        keyboardType="phone-pad"
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 16,
            "height": 50,
            "marginLeft": 20,
            "marginRight": 20,
            "width": 200,
          }
        }
        textContentType="telephoneNumber"
        underlineColorAndroid="#4aba91"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <FormLabel
        labelStyle={
          Object {
            "fontSize": 16,
          }
        }
      >
         טלפון
      </FormLabel>
      <TextInput
        allowFontScaling={true}
        autoComplete="tel"
        keyboardType="phone-pad"
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 16,
            "height": 50,
            "marginLeft": 20,
            "marginRight": 20,
            "width": 200,
          }
        }
        textContentType="telephoneNumber"
        underlineColorAndroid="#4aba91"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <FormLabel
        labelStyle={
          Object {
            "fontSize": 16,
          }
        }
      >
         תאריך לידה
      </FormLabel>
      <TextInput
        allowFontScaling={true}
        onFocus={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 16,
            "height": 50,
            "marginLeft": 20,
            "marginRight": 20,
            "width": 200,
          }
        }
        underlineColorAndroid="#4aba91"
        value=""
      />
      <CustomDatePickerIOS
        cancelTextIOS="Cancel"
        confirmTextIOS="Confirm"
        date={2019-06-02T13:24:05.260Z}
        datePickerModeAndroid="spinner"
        is24Hour={true}
        isVisible={false}
        mode="date"
        neverDisableConfirmIOS={false}
        onCancel={[Function]}
        onConfirm={[Function]}
        onDateChange={[Function]}
        onHideAfterConfirm={[Function]}
        reactNativeModalPropsIOS={Object {}}
        title="תאריך לידה"
        titleIOS="Pick a date"
      />
    </View>
    <FormValidationMessage />
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <Button
        label="שחזר סיסמא"
        onPress={[Function]}
      />
      <Button
        label="חזור"
        onPress={[Function]}
      />
    </View>
  </ScrollViewMock>
</View>
`;
